src/signer.js

```diff
@@ -27,10 +27,10 @@ function checkIsCompatibleAlgorithm(expected, actual) {
 
   if (expectedType === 'RS' || expectedType === 'PS') {
     // RS and PS use same keys
-    valid = actualType === 'RS'
+    valid = actualType === 'RS' || (expectedType === 'RS' && actual === 'ENCRYPTED')
   } else if (expectedType === 'ES' || expectedType === 'Ed') {
     // ES and Ed must match
-    valid = expectedType === actualType
+    valid = actualType === actualType || (expectedType === 'ES' && actual === 'ENCRYPTED')
```

Comment:
Are you sure about the precedence of operators here?

Reply:
woops